sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -101,11 +101,18 @@ def create_schema(self, schema_name: str, ignore_if_exists: bool = True) -> None
 
     def columns(self, table_name: TableName) -> t.Dict[str, exp.DataType]:
         """Fetches column names and types for the target table."""
+        from google.cloud.bigquery import TimePartitioningType
+
         table = self._get_table(table_name)
-        return {
+        columns = {
             field.name: exp.DataType.build(field.field_type, dialect=self.dialect)
             for field in table.schema
         }
+        if table.time_partitioning and not table.time_partitioning.field:
+            columns["_PARTITIONTIME"] = exp.DataType.build("TIMESTAMP")
+            if table.time_partitioning.type_ == TimePartitioningType.DAY:
```

Comment:
Is it the case that all pseudo column time partitioned tables have a `_PARTITIONTIME` column but possibly not a `_PARTITIONDATE` field? So there is no way to have a pseduo column that is date only?

Reply:
Yep, that is correct!

https://cloud.google.com/bigquery/docs/querying-partitioned-tables#query_an_ingestion-time_partitioned_table